eng/vcvars.ps1

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+if ($IsWindows) {
+    # find VS root
+    $vswhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    $visualStudioPath = & $vswhere -prerelease -latest -property installationPath
+    Write-Output "vs located at: $visualStudioPath"
+
+    # Call vcvars64.bat and write the set calls to file
+    cmd.exe /c "call `"$visualStudioPath\VC\Auxiliary\Build\vcvars64.bat`" && set > %temp%\vcvars.txt"
+
+    # Read the set calls and set the corresponding pwsh env vars
+    Get-Content "$Env:temp\vcvars.txt" | Foreach-Object {
+        if ($_ -match "^(.*?)=(.*)$") {
+            Set-Content "env:\$($matches[1])" $matches[2]
+            Write-Host "setting env: $($matches[1]) = $($matches[2])"
```

Comment:
Consider changing the order of these 2 lines, otherwise if `Set-Content` results in a crash then the previous output, shown by the moment of crash, will mislead.
```suggestion
            Write-Host "setting env: $($matches[1]) = $($matches[2])"
            Set-Content "env:\$($matches[1])" $matches[2]
```

Reply:
If set content crashes, the build will stop and nothing will leak into the current environment.